std/compress/lzss/backref.go

```diff
@@ -47,22 +47,32 @@ type backref struct {
 	bType   backrefType
 }
 
+func panicIfErr(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
 func (b *backref) writeTo(w *bitio.Writer, i int) {
-	w.TryWriteByte(b.bType.delimiter)
-	w.TryWriteBits(uint64(b.length-1), b.bType.nbBitsLength)
-	if b.bType.dictOnly {
-		w.TryWriteBits(uint64(b.address), b.bType.nbBitsAddress)
-	} else {
-		w.TryWriteBits(uint64(i-b.address-1), b.bType.nbBitsAddress)
+	panicIfErr(w.WriteByte(b.bType.delimiter))
```

Comment:
why the change? The purpose of the `TryWrite...` methods is to make things more readable and test the error afterwards -- probably also has a perf impact over a large stream to do so many branches at each write

Reply:
The errors overwrite each other so if we encounter an error in one of the middle writes (don't know how likely that is; to fail on one write and succeed on the following ones) we won't know that there ever was an error.